### ISSUE TITLE
A "guess" mode?

### ISSUE THREAD

**Issue by ghost** [Trixarian/sakis3g-source#8]
It would be great if sakis3g could try all devices and all detected APNs in order, trying to guess how to get online.

This would be useful for headless appliances like a Raspberry Pi hidden in a car, where you may need to swap out a USB 3G Dongle and connecting a monitor and keyboard and reconfiguring sakis3g is not be possible without disassembling the car.

**Comment by joshluongo** [comment]
While I'm not 100% sure as i have not used this in some time.

However i have found that it automatically finds the correct APN for my carriers when ever i have used it. Having a quick look at the source code it gets the settings from this file https://github.com/Trixarian/sakis3g-source/blob/master/files/operators.db

**Comment by tetherit** [comment]
It doesn't for me, it presents a list of 3 different APN settings and the first one works fine, however I need to manually specify a whole bunch of stuff:

sakis3g OTHER="USBMODEM" USBMODEM="12d1:1003" APN="internet" connect

If I don't specify OTHER, it dies asking what type of device this is. If I don't specify the USBMODEM, it just lists all the USB devices available and if I don't specify APN it lists 3 different APNs - it would be great if it could just try all the options it lists instead of showing them and dying.

**Comment by joshluongo** [comment]
Would i be correct in guessing that your are using a Huawei E220?

**Comment by tetherit** [comment]
That's right, now if only sakis3g would have a mode where it could attempt a guess too :P

**Comment by joshluongo** [comment]
I'm gonna look in to it, i think i have one of those modems somewhere. I mainly use a ZTE MF821 with it and it works fine. Another thing i have to ask is are you using the interface for it or do you use automated scripts?

**Comment by Trixarian** [comment]
Like this: http://web.archive.org/web/20120707165211/http://wiki.sakis3g.org/wiki/index.php?title=Sakis3G_configuration#voodoo ?

It still ends up asking the interface (for me anyway), but otherwise it's automated.

You can also invoke it like this: `sakis3g please connect` (please is an alias for --voodoo :wink:)

**Comment by Trixarian** [comment]
Another option could be to use [UMTSkeeper](http://mintakaconciencia.net/squares/umtskeeper/index.html) to automate it.

**Comment by tetherit** [comment]
Just a note, it seems the please connect or --voodoo option still asks for the APN :( -

```
$ sudo OTHER='USBMODEM' USBMODEM='12d1:1003' sakis3g please connect --voodoo
Please select APN by using APN variable, or by enabling interactive mode.
    $ /usr/local/bin/sakis3g --interactive "connect" "--voodoo"

Available options are:
internet    Contract (internet)
pp.vodafone.co.uk   Prepaid (pp.vodafone.co.uk)
pp.internet Old PAYG (pp.internet)
ppbundle.internet   New PAYG (ppbundle.internet)
CUSTOM_APN  Custom APN...

Example:
    $ /usr/local/bin/sakis3g APN="internet"
```

It would be great if sakis3g would just try each one to see if it can get online :) -- or at least provide an option to list the APNs in a more parseable format, like "sakis3g listapns"
